File: code/jvm/src/main/kotlin/pt/isel/leic/ptgest/domain/session/model/TrainerSessionDetails.kt
package pt.isel.leic.ptgest.domain.session.model

import pt.isel.leic.ptgest.domain.session.SessionType
import java.util.*

data class TrainerSessionDetails(
    val id: Int,
    val traineeName: String,
    val workoutId: Int,
    val beginDate: Date,
    val endDate: Date?,
    val location: String?,
    val type: SessionType,
    val notes: String?
)

File: code/js/src/views/user/TrainerViews/models/sessions/TrainerSessionDetails.ts
import SessionFeedback from "./SessionFeedback.ts";

class TrainerSessionDetails {
    id: number
    traineeName: string
    workoutId: number
    beginDate: string
    endDate: string | null
    location: string | null
    type: string
    notes: string | null
    cancelled: boolean = false
    reason: string | null
    source: string | null
    feedbacks: SessionFeedback[]

    constructor(
        id: number,
        traineeName: string,
        workoutId: number,
        beginDate: string,
        endDate: string | null,
        location: string | null,
        type: string,
        notes: string | null,
        cancelled: boolean,
        reason: string | null,
        source: string | null,
        feedbacks: SessionFeedback[]
    ) {
        this.id = id
        this.traineeName = traineeName
        this.workoutId = workoutId
        this.beginDate = beginDate
        this.endDate = endDate
        this.location = location
        this.type = type
        this.notes = notes
        this.cancelled = cancelled
        this.reason = reason
        this.source = source
        this.feedbacks = feedbacks
    }
}
export default TrainerSessionDetails
